### DISCUSSION TITLE
Tokens disappear?

### DISCUSSION THREAD

**Discussion by Textfield** [six7/figma-tokens#238] [Q&A]
Hey guys,

Anyone experiencing recent tokens just disappear from the plugin. I find myself re-adding the same tokens for 3-4 times. Even when pulling from JSONbin, it does not bring them back.

Is it a known issue? Is there a fix I can apply?

Thank you!

**Comment by six7** [comment]
Are you by chance creating tokens with  the same name? A name is unique and can only exist once. I think we'd need to add a check if that token exists yet (if we don't have that already).

Also, if not, anything that you can help with to reproduce this? Maybe even a screen recording? This would help tremendously.

**Reply by Textfield** [reply]
Thanks for the reply! 

Definitely not the same name. To give an example we have pink.100 - 900 shades which disappear even though that's the only pink color in our tokens. My suspicion is that, when 2 designers work on it at the same time, maybe a conflict between versions happen?

I'll try to record a video next time if this persists with JSONbin pulls too.

Besides that, amazing plug-in!

**Reply by six7** [reply]
Hm, there is no merging going on, if 2 people edit at the same time right now only of them wins, but it wouldnt merge them.

Please do! This is the first time I hear about this, I'd love to fix this asap if this is happening to you.

**Reply by badjar** [reply]
My colleagues and I are experiencing a similar issue. When I close and open the plugin some token names revert to earlier versions of their names. I've also seen some reordering of names. i.e. reverting to a different order.
Actually, the whole JSON code seems to revert to an earlier version.

I am working in an enterprise environment and other colleagues are collaborating on the same Figma file.  I have not tried to work out the steps to reproduce it, but it happens regularly.
My workaround is to take regular backups of the JSON code, but it is a bit of a hassle.

**Reply by six7** [reply]
@badjar what Sync are you using? Local, JSONBin or GitHub?

**Comment by badjar** [comment]
Currently just local storage. We're only working in one file for now.

I have a hunch the problems arise when I leave the plugin open in Figma when I put my computer to sleep. If someone else works with the plugin before I wake my computer I think the plugin has problems syncing the updates.

**Reply by phrutos** [reply]
Make sure the local storage folder where the tokens are hasn't been renamed. I was doing tests for my GitHub setups and I mixed the name of two folders, and the tokens wouldn't pull to figma. I updated the name back to "data/tokens.json" and boom. I hope that helps!
